test/extensions/filters/http/jwt_authn/authenticator_test.cc

```diff
@@ -251,6 +251,30 @@ TEST_F(AuthenticatorTest, TestSetHeader) {
       TestUtility::protoEqual(expected_payload, out_extracted_data_.fields().at("my_header")));
 }
 
+// This test verifies jwt status in failed status in metadata
+TEST_F(AuthenticatorTest, TestSetExpiredJwtToGetStatus) {
```

Comment:
Please add unit-test cases when  `allow_missing_or_failed` is used for
* missing
* failed
* two tokens:  one fail and one success
* two tokens:  both fails.
*